src/gfnff/gfnff_ini2.f90

```diff
@@ -776,83 +776,107 @@ subroutine gfnff_hbset(n,at,xyz,topo,neigh,nlist,hbthr1,hbthr2)
       integer i,j,k,nh,ia,ix,lin,ij,inh,jnh, iTri,iTrj,iTrDum
       real(wp) rab,rmsd, rih,rjh
       logical ijnonbond,free
+      ! local copies for omp parallelization
+      integer :: neigh_nTrans, neigh_numctr, nlist_nhb1, nlist_nhb2
 
+      neigh_nTrans = neigh%nTrans
+      neigh_numctr = neigh%numctr
 
       rmsd = sqrt(sum((xyz-nlist%hbrefgeo)**2))/dble(n)
 
       if(rmsd.lt.1.d-6 .or. rmsd.gt. 0.3d0) then ! update list if first call or substantial move occured
       nlist%nhb1=0
       nlist%nhb2=0
+      nlist_nhb1=0
+      nlist_nhb2=0
       ! loop over hb-relevant AB atoms
+      !$omp parallel do default(none) private(i, j, k, nh, ix, iTri, iTrj, iTrDum, rab, &
+      !$omp rih, rjh,ijnonbond,free ) &
+      !$omp firstprivate (nlist_nhb1, nlist_nhb2, neigh_numctr, neigh_nTrans, hbthr1, hbthr2) &
+      !$omp shared(nlist, topo, neigh, xyz)
```

Comment:
Maybe `schedule(dynamic)` improves performance here. The loop work could be different due to the conditions in the nested loops.

Reply:
Better to use `collapse(3)` with evaluation of i/j variables in inner loops to increase OpenMP iteration space. Playing with second arg of dynamic scheduling may also help: I usually use 32 or 64.